lib/beaker-puppet/install_utils/module_utils.rb

```diff
@@ -114,13 +114,16 @@ def install_puppet_module_via_pmt( opts = {} )
         def copy_module_to(one_or_more_hosts, opts = {})
           block_on one_or_more_hosts do |host|
             opts = {:source => './',
-                    :target_module_path => host['distmoduledir'],
                     :ignore_list => PUPPET_MODULE_INSTALL_IGNORE}.merge(opts)
 
+            target_module_dir = if opts[:target_module_path]
+                                  on( host, "echo #{opts[:target_module_path]}" ).stdout.chomp
+                                else
+                                  on( host, puppet("config print basemodulepath") ).stdout.chomp.split(':').first
```

Comment:
I have seen `puppet config` print some warning that could impact this in non-happy path scenarios:
```
$ puppet config print basemodulepath
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
/Users/$foo/modules:/opt/puppetlabs/puppet/modules
```
Which would end up copying to "WARN/$module_name"
It may be worth checking a valid path is captured before continuing with the copy. Or only process the last line of stdout?

Reply:
If Puppet is well behaved then it'll output warnings on stderr and it should be no problem but it's a good point. It could also show a deprecation warning. Checking if it's a valid path is a good suggestion and I'll look into this.

Reply:
Checking if the directory exists doesn't really work because this code is executed on the host, not on the client. There are also cases where a directory in the modulepath might not exist yet but can be created. I'd say that variable interpolation is not supported and needs to be passed in explicitly.